src/lib.rs

```diff
@@ -20,16 +24,37 @@ pub struct Config {
     pub redirect_url: String,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+// See https://tools.ietf.org/html/rfc6749#section-5.1
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Deserialize)]
 pub struct Token {
     pub access_token: String,
+    #[serde(default)]
     pub scopes: Vec<String>,
     pub token_type: String,
+    #[serde(default)]
+    pub expires_in: Option<u32>,
+    #[serde(default)]
+    pub refresh_token: Option<String>,
+}
+
+// See https://tools.ietf.org/html/rfc6749#section-5.2
+#[derive(Debug, Deserialize)]
+struct TokenError {
```

Comment:
I think due to the expansion of public fields in `Token` this'll need a major version bump anyway, so want to go ahead and expose this as a public type?